src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -206,15 +206,22 @@ private static Interop.Sys.OpenFlags PreOpenConfigurationFromOptions(FileMode mo
             {
                 default:
                 case FileMode.Open: // Open maps to the default behavior for open(...).  No flags needed.
-                case FileMode.Truncate: // We truncate the file after getting the lock
+                case FileMode.Truncate when !DisableFileLocking: // We truncate the file after getting the lock
```

Comment:
Rather than duplicating the case for Truncate and Create, I'd rather see this be something like:
```C#
case FileMode.Truncate:
    if (DisableFileLocking)
    {
        flags |= Interop.Sys.OpenFlags.O_TRUNC; // if we don't lock the file, we can truncate it when opening
    }
    break;
```
That way, everything about Truncate is contained in one case, and I don't have to go searching for all the different when combinations to figure out whether everything is covered.

Same for the other modes.
    }